src/lib.rs

```diff
@@ -369,8 +369,15 @@ impl<W: Write> Archive<W> {
     }
 
     fn append_path2(&self, path: &Path) -> io::Result<()> {
-        let mut file = try!(fs::File::open(path));
-        self.append_file(path, &mut file)
+        let stat = try!(fs::metadata(path));
+        if stat.is_file() {
+            let mut file = try!(fs::File::open(path));
+            self.append_file(path, &mut file)
```

Comment:
This may want to call `append` directly because the `Metadata` is already known (e.g. using `Header::set_metadata`)